src/EditorFeatures/CSharpTest/CodeActions/ConvertLinq/ConvertLinqQueryToForEachTests.cs

```diff
@@ -0,0 +1,2337 @@
+        {
+            string source = @"
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    List<int> M(IEnumerable<int> nums)
+    {
+        var list = new List<int>();
+        return ([|from int n1 in nums 
+                 from int n2 in nums
+                 select n1|]).ToList();
+    }
+}
+";
+
+            string output = @"
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    List<int> M(IEnumerable<int> nums)
+    {
+        var list = new List<int>();
+        var list1 = new List<int>();
+        foreach (int n1 in nums)
+        {
+            foreach (int n2 in nums)
+            {
+                list1.Add(n1);
+            }
+        }
+
+        return list1;
+    }
+}
+";
+
+            await TestInRegularAndScriptAsync(source, output);
+        }
+
+        [Fact]
+        public async Task AssignCount()
+        {
+            string source = @"
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    int M(IEnumerable<int> nums)
+    {
+        var cnt = ([|from int n1 in nums 
+                 from int n2 in nums
+                 select n1|]).Count();
+        return cnt;
+    }
+}
+";
+
+            string output = @"
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    int M(IEnumerable<int> nums)
+    {
+        var cnt = 0;
+        foreach (int n1 in nums)
+        {
+            foreach (int n2 in nums)
+            {
+                cnt++;
+            }
+        }
+
+        return cnt;
+    }
+}
+";
+
+            await TestInRegularAndScriptAsync(source, output);
+        }
+
+        [Fact]
+        public async Task ReturnCount()
+        {
+            string source = @"
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    int M(IEnumerable<int> nums)
+    {
+        return ([|from int n1 in nums 
+                 from int n2 in nums
+                 select n1|]).Count();
+    }
+}
+";
+
+            string output = @"
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    int M(IEnumerable<int> nums)
+    {
+        var count = 0;
+        foreach (int n1 in nums)
+        {
+            foreach (int n2 in nums)
+            {
+                count++;
+            }
+        }
+
+        return count;
+    }
+}
+";
+
+            await TestInRegularAndScriptAsync(source, output);
+        }
+
+        [Fact]
+        public async Task ReturnCountNameGeneration()
+        {
+            string source = @"
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    int M(IEnumerable<int> nums)
+    {
+        int count = 1;
+        return ([|from int n1 in nums 
+                 from int n2 in nums
+                 select n1|]).Count();
+    }
+}
+";
+
+            string output = @"
+using System.Collections.Generic;
+using System.Linq;
+class C
+{
+    int M(IEnumerable<int> nums)
+    {
+        int count = 1;
+        var count1 = 0;
```

Comment:
>count1 [](start = 12, length = 6)

For future PR, we should probably add a rename annotation as this is a new variable declaration.

Reply:
addressed [https://github.com/dotnet/roslyn/issues/25639](https://github.com/dotnet/roslyn/issues/25639)

---
In reply to: [176870335](https://github.com/dotnet/roslyn/pull/25362#discussion_r176870335) [](ancestors = 176870335)

Reply:
also use TypeStyleHelper for "var" so that it can follow code style options.

Reply:
I think we should create another annotation called TypeStyle so that we have post process that convert var to follow code style options.

also, we should have refactoring service that return document that changes code following code style options like

Document = ICodeStyleService.ApplyCodeStyle(Document, Options)

can you fie a bug for this? so that each refactoring, code fix doesn't need to care about those code style at all, and all features can follow all code styles as we add more of them?

Reply:
will be covered in [https://github.com/dotnet/roslyn/issues/25639](https://github.com/dotnet/roslyn/issues/25639)

---
In reply to: [176884216](https://github.com/dotnet/roslyn/pull/25362#discussion_r176884216) [](ancestors = 176884216)

Reply:
[https://github.com/dotnet/roslyn/issues/25694](https://github.com/dotnet/roslyn/issues/25694)

---
In reply to: [176884546](https://github.com/dotnet/roslyn/pull/25362#discussion_r176884546) [](ancestors = 176884546)